src/components/navigation-section.tsx

```diff
@@ -8,51 +8,41 @@ type Props = {
   section: NavigationItemList;
   currentSlug: string;
   onItemClick: () => void;
+  flatSections: NavigationSectionItem[];
 };
 
 const NavigationSection = ({
   title,
   section,
   currentSlug,
   onItemClick,
+  flatSections,
 }: Props) => {
   return (
     <ul className="side-nav__list">
       <h2 className="side-nav__title">{title}</h2>
       {section.map((item: NavigationSectionItem) => {
+        let flatItem: NavigationSectionItem = { ...item, isDone: false };
+        for (let i: number = 0; i < flatSections.length; i++) {
+          if (flatSections[i].slug === item.slug) {
+            flatItem = flatSections[i];
+            break;
+          }
+        }
```

Comment:
nit: This entire block can be simplified using the `find` method to just the below:

```diff
-        let flatItem: NavigationSectionItem = { ...item, isDone: false };
-        for (let i: number = 0; i < flatSections.length; i++) {
-          if (flatSections[i].slug === item.slug) {
-            flatItem = flatSections[i];
-            break;
-          }
-        }
+        const flatItem: NavigationSectionItem = flatSections.find(
+          flatSection => flatSection.slug === item.slug
+        ) || { ...item, isDone: false };
```

Reply:
@sagirk, I was using `.find` in until [4936852](https://github.com/nodejs/nodejs.dev/pull/187/commits/49368524625618e1ca5ef1b1c39e1021e11f2e9a), because I saw lack of IE support for [`.find` Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Browser_compatibility). Would like that declarative code as well, but think the IE desktop support may be still a concern, right?

Reply:
The `target` option in `tsconfig.json` should instruct the TS compiler to include the right polyfills in the generated dist bundles. Shouldn't be a concern (as of yet — we can always find the ones causing bloat and optimise later).

```js
{
  "compilerOptions": {
    //...,
    "target": "es5",
    //...,
}
```